coral/src/app/features/configuration/clusters/AddNewClusterForm.tsx

```diff
@@ -0,0 +1,199 @@
+import { Routes } from "src/app/router_utils";
+import { addNewCluster } from "src/domain/cluster";
+import { parseErrorMsg } from "src/services/mutation-utils";
+
+const AddNewClusterForm = () => {
+  const navigate = useNavigate();
+  const toast = useToast();
+  const form = useForm<AddNewClusterFormSchema>({
+    schema: addNewClusterFormSchema,
+    defaultValues: {
+      clusterType: "KAFKA",
+    },
+  });
+  const { clusterType, kafkaFlavor, clusterName } = form.watch();
+
+  // Because the protocol fields for the 'kafka' clusterType has more options than the other clusterTypes
+  // We need to reset the protocol field when the clusterType changes
+  useEffect(() => {
+    form.resetField("protocol");
+    return;
+  }, [clusterType]);
+
+  // Similarly, we need to reset the projectName and serviceName fields when the clusterType or kafkaFlavor changes
+  // As those fields should be undefined when the clusterType is not 'KAFKA' and the kafkaFlavor is not 'AIVEN_FOR_APACHE_KAFKA'
+  useEffect(() => {
+    form.resetField("projectName");
+    form.resetField("serviceName");
+    return;
+  }, [clusterType, kafkaFlavor]);
+
+  const addNewClusterMutation = useMutation(addNewCluster, {
+    onSuccess: () => {
+      navigate(`${Routes.CLUSTERS}?search=${clusterName}`);
+      toast({
+        message: "Cluster successfully added",
+        position: "bottom-left",
+        variant: "default",
+      });
+    },
+    onError: (error) => {
+      toast({
+        message: `Could not add cluster: ${parseErrorMsg(error)}`,
+        position: "bottom-left",
+        variant: "danger",
+      });
+    },
+  });
+
+  const onSubmitForm = (userInput: AddNewClusterFormSchema) => {
+    const formattedBootstrapServers = userInput.bootstrapServers.join(",");
+    const formattedAssociatedServers = userInput.associatedServers?.join(",");
+
+    addNewClusterMutation.mutate({
+      ...userInput,
+      bootstrapServers: formattedBootstrapServers,
+      associatedServers: formattedAssociatedServers,
+    });
+  };
+
+  return (
+    <Form {...form} ariaLabel={"Add new cluster"} onSubmit={onSubmitForm}>
+      <RadioButtonGroup<AddNewClusterFormSchema>
+        name="clusterType"
+        labelText="Cluster type"
+        required
+      >
+        <BaseRadioButton name="KAFKA" value="KAFKA">
+          Kafka
+        </BaseRadioButton>
+        <BaseRadioButton name="KAFKA_CONNECT" value="KAFKA_CONNECT">
+          Kafka Connect
+        </BaseRadioButton>
+        <BaseRadioButton name="SCHEMA_REGISTRY" value="SCHEMA_REGISTRY">
+          Schema registry
+        </BaseRadioButton>
+      </RadioButtonGroup>
+      <TextInput<AddNewClusterFormSchema>
+        name="clusterName"
+        labelText="Cluster name"
+        placeholder="DEV"
+        required
+      />
+
+      <NativeSelect<AddNewClusterFormSchema>
+        name="protocol"
+        labelText="Protocol"
+        placeholder="Select protocol"
```

Comment:
We tried to establish a pattern of using `-- Please select --` as a placeholder, because it's more different from the values itself and more clear that "this is something else".

Reply:
Sure ^^ [a62917b](https://github.com/Aiven-Open/klaw/pull/2301/commits/a62917b2c3ebf6645fa17b06e215fcce9c653c34)